sdk/storage/AzureStorageBlob/Source/NetworkingPolicy.swift

```diff
@@ -0,0 +1,58 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCore
+import Foundation
+
+/// Describes the type of network connection
+public enum NetworkType {
+    case unknown
```

Comment:
I think we probably need to have two different enums, and map between them:

1.) The network types that the developer can choose to allow or deny transfers on. This only has two values, wifiOrEthernet and cellular, and is public.
2.) The current network state. This has all four values and is internal. We use this to keep track of the current network state.

Mostly because we don't want a developer to be able to say "transfer only over `disconnected`", that doesn't make sense. But we want to know when we are going from a connected network to `disconnected` or vice versa.

Reply:
Done.

Reply:
Also, not sure if this is how you implemented it or not, but for #1, this should probably be an OptionSet rather than a list of enum values.

Reply:
I converted both to an OptionSet.